src/core/internal/array/equality.d

```diff
@@ -203,3 +126,48 @@ bool __equals(T1, T2)(T1[] lhs, T2[] rhs)
     assert(!__equals(a1, a2));
     assert(a1 != a2);
 }
+
+
+private:
+
+template isVoid(T)
+{
+    import core.internal.traits : Unqual;
+    enum isVoid = is(Unqual!T == void);
```

Comment:
From the perspective of the compiler frontend there anything to recommend this over `is(immutable T == immutable void)`?

Reply:
What about `shared`?

Reply:
It works since `immutable(shared T)` == `immutable T`.

Reply:
> From the perspective of the compiler frontend there anything to recommend this over is(immutable T == immutable void)?

Absolutely not, thx for the suggestion.